src/Security/Authentication/Cookies/src/CookieAuthenticationHandler.cs

```diff
@@ -303,9 +303,23 @@ protected async override Task HandleSignInAsync(ClaimsPrincipal user, Authentica
             {
                 if (_sessionKey != null)
                 {
-                    await Options.SessionStore.RemoveAsync(_sessionKey);
+                    // Renew the ticket in cases of multiple requests see: https://github.com/dotnet/aspnetcore/issues/22135
+                    var oldTicket = await Options.SessionStore.RetrieveAsync(_sessionKey);
```

Comment:
I don't think you need to call Retrieve again, it was already called over here:
https://github.com/dotnet/aspnetcore/blob/fcb43d3b3b18a6dc76f1b32beffc14bb4179670e/src/Security/Authentication/Cookies/src/CookieAuthenticationHandler.cs#L143-L148

When we call Retrieve the first time, delay setting _sessionKey until after we see Retrieve succeed.

Reply:
Okay, so we rely on sessionKey to determine if we renew or remove, much cleaner too